spec/requests/users/registrations/check_details_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Users::Registrations /check-details", type: :request do
+  let(:school) { FactoryBot.create(:school) }
+  let(:email) { Faker::Internet.email(domain: school.domains.first) }
+
+  describe "POST /users/check_email" do
+    it "renders correct template when no schools are found" do
```

Comment:
Rather than 'correct template' in many places in this file, I would expect the names of templates in the test description - so that when someone has a failing test they won't be tempted to just quickly change the template in it.